src/axis.tsx

```diff
@@ -1,31 +1,22 @@
 import { AxisStore, IAxisStoreOptions } from "./store";
-import { AxisDataType, Color, Vec2, Vec3 } from "./types";
+import { Vec2, Vec3 } from "./types";
 
-export interface IAxisProps extends IAxisStoreOptions {
-
-}
+export interface IAxisProps extends IAxisStoreOptions { }
 
 export class Axis {
   store: AxisStore;
 
   constructor(props: IAxisProps) {
     this.store = new AxisStore({
-      padding: {
-        left: props.padding.left,
-        right: props.padding.right,
-        top: props.padding.top,
-        bottom: props.padding.bottom
-      },
+      origin: props.origin,
+      width: props.width,
+      height: props.height,
```

Comment:
Instead of splitting out the properties of the viewbox can you just make it a single property to be set? Like:

```
view: {
  x: number;
  y: number;
  width: number;
  height: number;
}
```

or even:

```
view: {
  origin: Vec2;
  size: Vec2;
}
```